docs/src/DeveloperDocumentation/styleguide.md

```diff
@@ -190,6 +190,54 @@ end
 However, as always, rules sometimes should be broken.
 
 
+## Optional arguments for parents of return values
+
+Several objects in OSCAR have `parent`s, e.g. polynomials, group elements, ... 
+Whenever a function creates such objects from an input which does not involve 
+the output's parent, the user should have the possibility to pass on this 
+parent as an optional argument. 
+
+Let's see an example. Say, you want to implement the characteristic 
+polynomial of a matrix. You could do it as follows:
+```julia
+  function characteristic_polynomial(A::MatrixElem)
+    kk = base_ring(A)
+    P, x = kk["x"]
```

Comment:
```suggestion
    P, x = kk[:x]
```

Reply:
both are terrible - and should be replaced by `polynomial_ring(kk, cached = false)`. Within the system, `kk[:x]` and variants should never be used as they create global cached objects that clutter up the memory.